src/util/animation/AnimationFrameProvider.ts

```diff
@@ -1,23 +1,28 @@
 import { getEnv } from '../../env';
 
-const _requestAnimFrame: AnimationFrameProvider['requestAnimationFrame'] =
-  getEnv().window.requestAnimationFrame ||
-  function (callback: FrameRequestCallback) {
-    return getEnv().window.setTimeout(callback, 1000 / 60);
-  };
-
-const _cancelAnimFrame: AnimationFrameProvider['cancelAnimationFrame'] =
-  getEnv().window.cancelAnimationFrame || getEnv().window.clearTimeout;
+let _requestAnimFrame: AnimationFrameProvider['requestAnimationFrame'];
+let _cancelAnimFrame: AnimationFrameProvider['cancelAnimationFrame'];
 
 /**
  * requestAnimationFrame polyfill based on http://paulirish.com/2011/requestanimationframe-for-smart-animating/
  * In order to get a precise start time, `requestAnimFrame` should be called as an entry into the method
  * @param {Function} callback Callback to invoke
  */
 export function requestAnimFrame(callback: FrameRequestCallback): number {
+  if (!_requestAnimFrame) {
+    _requestAnimFrame =
+      getEnv().window.requestAnimationFrame ||
+      function requestAnimationFramePolyfill(callback: FrameRequestCallback) {
+        return getEnv().window.setTimeout(callback, 1000 / 60);
+      };
+  }
   return _requestAnimFrame.call(getEnv().window, callback);
 }
 
 export function cancelAnimFrame(handle: number): void {
+  if (!_cancelAnimFrame) {
+    _cancelAnimFrame =
+      getEnv().window.cancelAnimationFrame || getEnv().window.clearTimeout;
+  }
```

Comment:
in another PR i want to come back over this file.
I want to know with current browser support and current JSDOM who does or does not support request animation frame and if we still need polyfills.